src/compute-client/src/controller.rs

```diff
@@ -830,11 +832,15 @@ where
                 write_only: dataflow.sink_exports.contains_key(&id),
                 compute_dependencies: dataflow.imported_index_ids().collect(),
                 shared: shared.clone(),
+                time_dependence: time_dependence.clone(),
             };
             instance.collections.insert(id, collection);
             shared_collection_state.insert(id, shared);
         }
 
+        let mut dataflow = dataflow;
```

Comment:
Nit: I'd just add the `mut` to the `dataflow` parameter instead of having this extra reassignment.